x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -228,6 +221,8 @@ export const searchAfterAndBulkCreate = async ({
               tuple.maxSignals - signalsCreatedCount
             );
           }
+          const enrichedEvents = await enrichment(filteredEvents);
```

Comment:
Just observing here: @rylnd I think the big loop will exit early here if the enrichment function throws an error. Am I understanding that correctly?

Reply:
@dhurley14 that's correct. The default function here won't throw, but for indicator match rules if they have e.g. [a primitive value instead of nothing/an object](https://github.com/elastic/kibana/pull/89899/files#diff-dd470b1efd3f96e29ea2aeddee97b7233e1412f6a1374fc821efdb3b9235807fR399), the signal creation would exit with an error. We could certainly try/catch this and continue to create signals without enrichments, but I'm not sure what the preferred behavior should be here. @FrankHassanabad @spong thoughts?

I'll capture whatever behavior we decide here with an integration test.